core/src/main/java/org/axonframework/serialization/AbstractXStreamSerializer.java

```diff
@@ -145,6 +146,16 @@ protected AbstractXStreamSerializer(Charset charset, XStream xStream, RevisionRe
 
         xStream.alias("meta-data", MetaData.class);
         xStream.registerConverter(new MetaDataConverter(xStream.getMapper()));
+
+        // Make sure ClassLoader of this instance is registered - required by spring devtools RestartClassLoader
+        if (CompositeClassLoader.class.isAssignableFrom(xStream.getClassLoader().getClass())) {
+            ((CompositeClassLoader)xStream.getClassLoader()).add(this.getClass().getClassLoader());
+        } else {
+            final CompositeClassLoader compositeClassLoader = new CompositeClassLoader();
```

Comment:
If the already configured class loader is not a CompositeClassLoader, probably we want to register it with the CompositeClassLoader we're creating. It might have been configured for a reason 😉